src/test/controller/adverse.spec.ts

```diff
@@ -0,0 +1,174 @@
+        }
+        if(addressId){
+            await Address.findByIdAndRemove(addressId);
+        }
+        if(adverseId){
+            await Adverse.findByIdAndRemove(adverseId);
+        }
+    });
+    describe('Get all Adverse list', () => {
+        it('should return 401 unauthorized without a valid session token', async () => {
+          const res = await chai
+            .request(app)
+            .get('/adverse/');
+            expect(res.body).to.have.property('message', 'Not authenticated.')
+        });
+        it('should get all adverse', async () => {
+          const res = await chai
+            .request(app)
+            .get('/adverse/')
+            .set('Authorization', `Bearer ${token}`);
+          expect(res).to.have.status(200);
+          expect(res.body).to.have.property('message', 'Adverse action list fetched Successfully');
+          expect(res.body).to.have.property('data').to.be.an('array');
+        });
+      });
+
+
+  describe(" Add Adverse list", () => {
+    it("should create a new candidate", async () => {
+    const address = new Address({
+        houseNo: "1",
+        streetNo: "456",
+        city: "City",
+        state: "State",
+        pincode: "123456",
+      });
+      await address.save();      
+      addressId = address._id;
+      const candidate = new Candidate({
+        firstName: "sam",
+        lastName: "last",
+        email: "sam@example.com",
+        contact: "9867546324",
+        license: "ABC123",
+        DOB: "1990-01-01",
+        address: address._id,
+        recruiter: id,
+      });
+      await candidate.save();
+      candidateId = candidate?._id as Types.ObjectId
+      const candidateDetails = await Candidate.find({candidate: candidateId});
+
+      const res = await chai
+        .request(app)
+        .post(`/adverse/${candidateId}`)
+        .set('Authorization', `Bearer ${token}`)
+        .send({
+            name:'sam',
+            status:'clear',
+            pre_notice_date:'1990-01-01',
+            post_notice_date:'1990-02-02',
+            candidate:candidateId
+        });
+        adverseId = res.body.Report._id;
+      expect(res).to.have.status(201);
+      expect(res.body).to.be.an("object");
+      expect(res.body.message).to.equal("Candidate adverse Report added successfully");
+
+    });
+    it("should throw error if adverse found", async () => {
+        const address = new Address({
+            houseNo: "1",
+            streetNo: "456",
+            city: "City",
+            state: "State",
+            pincode: "123456",
+          });
+          await address.save();      
+          addressId = address._id;
+          const candidate = new Candidate({
+            firstName: "sam",
+            lastName: "last",
+            email: "sam@example.com",
+            contact: "9867546324",
+            license: "ABC123",
+            DOB: "1990-01-01",
+            address: address._id,
+            recruiter: id,
```

Comment:
use a common model for testing data and import them. you will be able to reuse them